tools/CouchDB_MRI_Importer.php

```diff
@@ -96,24 +96,82 @@ function _generateCandidatesQuery($ScanTypes)
+            // exists, the subselect will return 'No selected file'. Otherwise the     //
+            // file path of the selected file is returned by the subselect             //
+            //-------------------------------------------------------------------------//
+            $Query .= ', '
+                    . '('
+                    . ' CASE ('. $this->_getQueryForSelectedFiles('count(*)', $scantype) . ') '
+                    . '  WHEN 1 '
+                    . '    THEN (' . $this->_getQueryForSelectedFiles('f.File', $scantype) . ') '
+                    . '  WHEN 0 '
+                    . '    THEN "No selected file" '
+                    . '  ELSE "Multiple Selected files"'
+                    . ' END'
+                    . ") as Selected_$scantype";
+
+            //---------------------------------------------------------------------------//
+            // Add to the query a subselect that will compute the Qc status of the       //
+            // selected file for the given session/modality. If more than one selected   //
+            // file exists, the subquery will return 'Unknown: multiple Selected files'. //
+            // If no selected file exists, the subselect will return 'No selected file'. //
+            // Otherwise the subselect will return the Qc status of the selected file (or//
+            // 'No Qc on selected file' if the file has not been Qced)                   //
+            //---------------------------------------------------------------------------//
+            $Query  .= ', '
+                    . '('
+                    . ' CASE (' . $this->_getQueryForSelectedFiles('count(*)', $scantype) . ') '
+                    . '  WHEN 1 '
+                    . '    THEN ('
+                    .        $this->_getQueryForSelectedFiles(
+                                 'COALESCE(fqs.QCStatus, "No Qc on selected file")',
+                                 $scantype,
+                                 's.ID'
+                             )
+                    .     ') '
+                    . '  WHEN 0 '
+                    . '    THEN "No selected file" '
+                    . '  ELSE "Unknown: multiple Selected files"'
+                    . ' END'
+                    . ") as {$scantype}_QCStatus";
         }
-        $Query .= " FROM session s JOIN candidate c USING (CandID)
-            LEFT JOIN feedback_mri_comments fmric
-            ON (fmric.CommentTypeID=7 AND fmric.SessionID=s.ID)
-            WHERE c.Entity_type != 'Scanner' AND c.PSCID NOT LIKE '%9999'
-                  AND c.Active='Y' AND s.Active='Y' AND s.CenterID <> 1";
+
+        $Query .= " FROM session s"
+                . " JOIN candidate c USING (CandID)"
+                . " LEFT JOIN feedback_mri_comments fmric"
+                . " ON (fmric.CommentTypeID=7 AND fmric.SessionID=s.ID)"
+                . " WHERE c.Entity_type != 'Scanner' AND c.PSCID NOT LIKE '%9999'"
```

Comment:
I have no idea why we reject the pscids that are like `'%9999'`...You?

Reply:
fascinating... In PREVENT-AD we had a candidate MTL9999 that needed to be excluded from the queries... Could this be some leftover of PREVENT-AD? LOL